### PR TITLE
Add filter pills to quests corresponding to in-game quests tabs

### PR DESCRIPTION
**Pull request by robojumper**
grafik

### CODE REVIEW

File: src/app/progress/Pursuits.tsx [hunk 1]
```diff
@@ -25,7 +26,16 @@ export const sortPursuits = chainComparator(
   compareBy((item) => item.name)
 );
 
-const pursuitsOrder = ['Bounties', 'Quests', 'Items'];
+const pursuitsOrder = ['Bounties', 'Quests', 'Items'] as const;
+
+// FIXME use d2ai trait hashes
```
Comment (bhollis) [inline src/app/progress/Pursuits.tsx:31]: This has merged now

File: src/app/progress/Pursuits.tsx [hunk 2]
```diff
@@ -93,16 +111,56 @@ export function PursuitsGroup({
...
+      _.compact(
+        pursuitCategoryTraitHashes.map((hash) => {
+          const def = defs.Trait.get(hash);
+          return def && { key: hash, content: def.displayProperties.name };
+        })
+      ),
+    [defs.Trait]
+  );
+
+  const [selectedFilters, setSelectedFilters] = useState<Option<number>[]>([]);
```
Comment (bhollis) [inline src/app/progress/Pursuits.tsx:153]: Is it problematic to have two sets of state for selected filters? For example, can you not shift-select both quest type and other bounty guide pills?

Comment (robojumper) [reply src/app/progress/Pursuits.tsx:153]: It's not hugely problematic -- they're just two different rows of buttons -- but it's probably a bit confusing. I can try integrating it into the existing bounty guide.

Comment (robojumper) [reply src/app/progress/Pursuits.tsx]: I moved it to the existing bounty guide and it's probably better that way. Also included the trait icons for some flavor.

File: src/app/progress/BountyGuide.tsx [hunk 3]
```diff
@@ -260,6 +279,17 @@ function PillContent({
           {defs.InventoryItem.get(value).displayProperties.name}
         </>
       );
+    case 'QuestTrait':
+      return (
+        <>
+          {/* the seasonal quest trait has the Season of the Lost icon? */}
+          {value !== TraitHashes.Seasonal_Quests &&
+            defs.Trait.get(value).displayProperties.hasIcon && (
```
Comment (bhollis) [inline src/app/progress/BountyGuide.tsx:287]: Would be nice to extract `defs.Trait.get(value).displayProperties` into a local

**Review by bhollis — APPROVED** [review]

**Review by bhollis — APPROVED** [review]

**Pull request merged**